Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift

```diff
@@ -423,10 +419,22 @@ open class ChatChannelListViewModel: ObservableObject, ChatChannelListController
             return
         }
 
-        searchResults = messageSearchController.messages
-            .compactMap { message in
-                message.makeChannelSelectionInfo(with: chatClient)
+        queue.async {
+            let results: [ChannelSelectionInfo] = messageSearchController.messages.compactMap { message in
+                guard let channelId = message.cid else { return nil }
+                guard let channel = messageSearchController.client.channelController(for: channelId).channel else {
```

Comment:
why do we have to go through the message search controller here? We have the client injected in the VM.

Reply:
ah yes you are correct 👍

Reply:
Done ✅